### PR TITLE
TST: troubleshoot strict=False xfails

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/computation/test_eval.py [hunk 1]
```diff
@@ -860,19 +860,6 @@ def test_basic_frame_series_alignment(
     def test_basic_series_frame_alignment(
         self, request, engine, parser, index_name, r_idx_type, c_idx_type
     ):
-        if (
```
Comment (mroeschke) [inline pandas/tests/computation/test_eval.py:863]: Maybe fixing a random seed might help make this more deterministic?

File: pandas/tests/extension/json/test_json.py [hunk 2]
```diff
@@ -276,7 +276,6 @@ def test_combine_add(self, data_repeated):
     @pytest.mark.xfail(
         reason="combine for JSONArray not supported - "
         "may pass depending on random data",
-        strict=False,
```
Comment (mroeschke) [inline pandas/tests/extension/json/test_json.py:279]: Same here; maybe the random seed should be fixed in the data fixture in `test_json.py`

File: pandas/tests/computation/test_eval.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/computation/test_eval.py:863]: i think that would work, but in both this case and the JSONArray case there may be an actual bug in play, so we would want to have an xfailed test?

Comment (mroeschke) [reply pandas/tests/computation/test_eval.py:863]: Fair point. From https://github.com/pandas-dev/pandas/pull/46796 for example, probably best if its xfailed even if strict=False due to some seemingly random sorting bug

**Comment by jbrockmendel** [comment]
dont merge, i want to re-run a few times to see if the flaky fails come back

**Review by mroeschke — APPROVED** [review]
LGTM. Merge when ready

**Pull request merged**